lir/algorithms/bootstraps.py

```diff
@@ -0,0 +1,229 @@
+
+
+class BootstrapEquidistant(Pipeline):
+    """Bootstrap system that estimates confidence intervals around the best estimate of a pipeline.
+
+    This bootstrap system creates bootstrap samples from the training data, fits the pipeline on each sample,
+    and then computes confidence intervals for the pipeline outputs based on the variability across the bootstrap
+    samples.
+
+    Computing these intervals is done by creating interpolation functions that map the best estimate to the
+    difference between the best estimate and the lower and upper bounds of the confidence interval. This interpolation
+    is fitted on equidistant points after computing the bootstrap estimates. In contrast to the BootstrapAtData class,
+    this class requires the creation of equidistant points for interval estimation. This means the input data must be
+    of a type that can be represented as equidistant points (e.g., continuous features).
+
+    Attributes:
+        interval: The lower and upper quantiles for the confidence interval.
+        n_bootstraps: The number of bootstrap samples to generate.
+        f_delta_interval_lower: Interpolation function for the lower bound of the interval.
+        f_delta_interval_upper: Interpolation function for the upper bound of the interval.
+
+    """
+
+    def __init__(
+        self,
+        steps: list[tuple[str, Any]],
+        n_bootstraps: int = 400,
+        interval: tuple[float, float] = (0.05, 0.95),
+        seed: int | None = None,
+        n_points: int | None = 1000,
+    ):
+        """Initialize the TrainDataBootstrap with the given pipeline steps, number of bootstraps, and interval.
+
+        Parameters:
+        param steps: list[tuple[str, Any]]: The steps of the pipeline to be bootstrapped.
+        param n_bootstraps: int: The number of bootstrap samples to generate. Default is 400.
+        param interval: tuple[float, float]: The lower and upper quantiles for the confidence interval.
+                                             Default: (0.05,0.95).
+        param n_points: int | None: The number of equidistant points to use for interval estimation. Default is 1000.
+                                    If None, uses the number of instances in the training data.
+        """
+        self.interval = interval
+        self.n_bootstraps = n_bootstraps
+        self.seed = seed
+        self.n_points = n_points
+        self.f_delta_interval_lower = None
+        self.f_delta_interval_upper = None
+
+        super().__init__(steps)
+
+    def fit(self, instances: FeatureData) -> Self:
+        """Fit the bootstrap system to the provided instances.
+
+        param instances: FeatureData: The feature data to fit the bootstrap system on.
+        return Self: The fitted bootstrap system.
+        """
+
+        all_vals = []
+        rng = np.random.default_rng(self.seed)
+
+        # Raise an exception if the dimensions are not what we expect.
+        if instances.features.ndim != 2:
```

Comment:
Deze test doet maar de helft, want wat je wil is dat hij 1 kolom heeft.

```suggestion
        if instances.features.ndim != 2 or instances.features.shape[1] != 1:
            raise ValueError(f'expected 2D feature array with 1 column; found shape: {instances.features.shape}')
```